src/packageDocs/java/net/minecraft/nbt/package-info.java

```diff
@@ -0,0 +1,234 @@
+ * are arrays internally, they also support {@linkplain AbstractNbtList#add adding items}
+ * at runtime. Like Java arrays, out of bounds access is forbidden and will throw
+ * {@link ArrayIndexOutOfBoundsException}.
+ *
+ * <h3 id=nbt-list>{@link NbtList}</h3>
+ * <p>NbtList is a list of a specific type of NBT elements. Empty lists always have the type
+ * set as {@link NbtEnd}. {@link NbtList#add} throws when the type is incompatible, while
+ * {@link NbtList#addElement} does nothing and returns {@code false} in that case.
+ * There are type-specific getters which can be used to get the value. They return the default
+ * value when type mismatch occurs or when the index is out of bounds. Note that they do not
+ * cast the value at all.
+ *
+ * <pre>{@code
+ *   NbtList list = new NbtList();
+ *   list.addElement(NbtFloat.of(0.5f));
+ *   float firstItem = list.getFloat(0); // 0.5f
+ *   float secondItem = list.getFloat(1); // 0.0f (default value)
+ *   double firstItemDouble = list.getDouble(0); // 0.0 (no casting)
+ *   list.addElement(NbtDouble.of(1.0)); // returns false, does nothing
+ *   list.add(NbtDouble.of(2.0)); // throws UnsupportedOperationException
+ * }</pre>
+ *
+ * <h3 id=nbt-compound>{@link NbtCompound}</h3>
+ * <p>NbtCompound is a hash map-like key-value storage object. It also acts as the root
+ * object for serialized NBTs. The keys are always strings, while the values can be of
+ * any type, and multiple value types can be mixed in a single compound. The order of items
+ * is not guaranteed. There are generic {@link NbtCompound#put} and {@link NbtCompound#get}
+ * methods, and there also exist type-specific getters and putters. Like the list getters
+ * mentioned above, they return the default value when the key is missing or type mismatch
+ * occurs. However, unlike lists, they attempt to cast numeric values.
+ *
+ * <pre>{@code
+ *   NbtCompound compound = new NbtCompound();
+ *   compound.put("Awesomeness", NbtInt.of(90));
+ *   // Don't do this! This will crash if the type is incompatible,
+ *   // even if they can be casted!
+ *   NbtInt awesomenessNbt = (NbtInt)compound.get("Awesomeness");
+ *   compound.putLong("Awesomeness", 100L);
+ *   int awesomeness = compound.getInt("Awesomeness"); // 100 (after casting)
+ *   int evilness = compound.getInt("Evilness"); // 0 (default value)
+ *   // Shortcuts for getting and putting a boolean value
+ *   // (internally represented as a byte)
+ *   compound.putBoolean("Awesome", true);
+ *   boolean awesome = compound.getBoolean("Awesome");
+*    compound.put("awesome", NbtDouble.of(1.0)); // key is case-sensitive
+ * }</pre>
+ *
+ * <h2 id=using-nbt>Using NBT</h2>
+ * <p>As noted before, NBT is a <strong>serialized format, not runtime format</strong>.
+ * Therefore, modifying NBT is usually not the solution for attaching custom data to
+ * objects like entities. For this purpose it is recommended to use third-party APIs
```

Comment:
Mention that it's not a solution as the custom NBT will be lost once processed by the game.